src/async_fuse/memfs/direntry.rs

```diff
@@ -0,0 +1,128 @@
+// Use of external crates and modules
+use datenlord::common::error::DatenLordError;
+use nix::sys::stat::SFlag;
+use serde::{Deserialize, Serialize};
+use tracing::warn;
+
+use crate::async_fuse::fuse::protocol::INum;
+
+/// Represents the type of a file in a filesystem.
+///
+/// This enum is used to distinguish between directories, files, and symlinks.
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
+pub enum FileType {
+    /// A directory.
+    Dir,
+    /// A regular file.
+    File,
+    /// A symbolic link.
+    Symlink,
+}
+
+impl TryFrom<SFlag> for FileType {
+    type Error = DatenLordError;
+
+    /// Attempts to convert an `SFlag` value into a `FileType`.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - The `SFlag` value representing file type at the OS level.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(FileType)` - If the conversion is successful.
+    /// * `Err(DatenLordError)` - If the `SFlag` value does not correspond to a
+    ///   known `FileType`.
+    fn try_from(value: SFlag) -> Result<Self, Self::Error> {
+        match value {
+            SFlag::S_IFDIR => Ok(Self::Dir),
+            SFlag::S_IFREG => Ok(Self::File),
+            SFlag::S_IFLNK => Ok(Self::Symlink),
+            _ => {
+                warn!("Try convert {:?} to FileType failed.", value);
+                Err(DatenLordError::ArgumentInvalid { context: vec![] })
+            }
+        }
+    }
+}
+
+/// Represents a directory entry in a filesystem.
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct DirEntry {
+    /// The inode number of the child
+    inum: INum,
+    /// The name of the child
+    name: String,
+    /// The type of the child
+    file_type: FileType,
+}
+
+impl DirEntry {
+    /// Creates a new `DirEntry`.
+    ///
+    /// # Arguments
+    ///
+    /// * `inum` - The inode number of the file or directory.
+    /// * `name` - The name of the file or directory.
+    /// * `file_type` - The type of the file (directory, file, or symlink).
+    ///
+    /// # Returns
+    ///
+    /// * `DirEntry` - The new `DirEntry` instance.
+    #[must_use]
+    pub fn new(inum: INum, name: String, file_type: FileType) -> Self {
+        Self {
+            inum,
+            name,
+            file_type,
+        }
+    }
+
+    /// Returns the inode number of the file or directory.
+    #[must_use]
+    pub fn get_inum(&self) -> INum {
+        self.inum
+    }
+
+    /// Returns the name of the file or directory.
+    #[must_use]
+    pub fn get_name(&self) -> &str {
+        &self.name
+    }
+
+    /// Returns the type of the file (directory, file, or symlink).
+    #[must_use]
+    pub fn get_file_type(&self) -> FileType {
+        self.file_type.clone()
+    }
```

Comment:
Getters in Rust are not with the prefix `get_` conventionally, just name them as `inum()`, `name()` and `file_type()`.

Reply:
Ok, I'll fix it.